.codeclimate.yml

```diff
@@ -4,53 +4,114 @@
 # codeclimate-common repo at https://github.com/kapost/codeclimate-common
 
 ---
-engines:
+version: 2
+plugins:
   rubocop:
     enabled: true
   bundler-audit:
     enabled: true
   duplication:
     enabled: true
+    exclude_patterns:
+    - "**/*Spec.js"
+    - "**/*_spec.rb"
+    - "**/specs/"
+    - "**/spec/"
+    - "spec/"
+    - "**/tests/"
+    - "**/test/"
+    - "test/"
+    - "features/"
+    - "lib/tasks/"
+    - "tasks/"
     config:
       languages:
-        - ruby
-        - javascript
-    exclude_paths:
-      - "spec/"
+        javascript:
+          mass_threshold: 140
+        ruby:
+          mass_threshold: 25 # default in source code as of 01/18/2018 (docs out of date)
   reek:
     enabled: true
   shellcheck:
     enabled: true
-  brakeman:
-    enabled: true
   fixme:
     enabled: true
-  markdownlint:
-    enabled: true
+    config:
+      strings:
+      - FIXME
   csslint:
-    enabled: true
+    enabled: false
   scss-lint:
     enabled: true
   coffeelint:
-    enabled: true
+    enabled: false
   eslint:
     enabled: true
-  # Uncomment this one if your app has an npm-shrinkwrap
-  # nodesecurity:
-  #   enabled: true
-
-ratings:
-  paths:
-    - app/**
-    - lib/**
-    - "**/*.rb"
-    - "**/*.js"
-    - "**/*.coffee"
-    - Gemfile.lock # For bundler-audit
-
-exclude_paths:
-  - "**/vendor/**/*"
-  - "**/node_modules/**/*"
-  - "spec/fixtures/**/*"
-
+    channel: "eslint-4"
+    config:
+      extensions:
+      - .es6
+      - .js
+      - .jsx
+    checks:
+      keyword-spacing:
+        enabled: false
+      react/jsx-space-before-closing:
+        enabled: false
+      no-warning-comments: # these are nice for client
+        enabled: false
+  nodesecurity:
+    enabled: false # turn back on if they start supporting yarn.lock
+  grep:
+    enabled: true
+    config:
+      patterns:
+        no-dot-only:
+          pattern: \.only
+          annotation: "Make sure not to commit `.only` in specs, as it disables the rest of the suite."
+          severity: blocker
+          path_patterns:
+          - "**/*_spec.rb"
```

Comment:
I think this is a JavaScript only thing.  There are usages of `.only` all over the Ruby specs in napa.

Reply:
Ah right—because of the activerecord method. I'll update to warn on `it.only` and `describe.only`, as that's something that can mess up rspec suites as well as JS

Reply:
I'm not sure you can do that in rspec?  I tried and got this `undefined method `only'`.

Reply:
ah I don't know why I thought that was a thing. I guess I can check for `fit` https://stackoverflow.com/a/22113489/1493191 but I don't know if anyone uses it

Reply:
also `focus: true` or `:focus => true

Reply:
Yes, those ☝️

Reply:
I'm good with simply taking `**/*_spec.rb` out of this rule.